src/main/java/reposense/parser/MarkdownParser.java

```diff
@@ -0,0 +1,34 @@
+package reposense.parser;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+import reposense.parser.exceptions.InvalidMarkdownException;
+import reposense.system.LogsManager;
+
+/**
+ * Parses Markdown file according to the "<!--repo-->" tag.
+ *
+ * @param <T> Generic Type {@code T}.
+ */
+public abstract class MarkdownParser<T> {
+    protected static final Pattern DELIMITER = Pattern.compile("<!--repo-->(.*)");
```

Comment:
Should this `DELIMTER` pattern be stored in the `BlurbMarkdownParser` instead? It seems specific to the blurbs use case?

Reply:
I wanted to follow the pattern used in `CsvParser` whereby the delimiter is stored in the abstract class, but perhaps I can move it to the concrete class to reduce coupling?

Reply:
Yes, I believe the other concrete MarkdownParsers will have different delimiter patterns, eg. `"<!--author-->(.*)"` for blurbs for authors, so we can decouple them. @gok99

Reply:
Sure thing, will refactor!